integration_tests/kpi.feature

```diff
@@ -56,12 +56,18 @@ Feature: KPI integration tests
      Then self_sufficiency of {'Grid': 1.0, 'Community': 0.666, 'House 1': 0.0, 'House 2': 0.5, 'House 3': 1.0} are correctly reported
      And self_consumption of {'Grid': 1.0, 'Community': 0.8, 'House 1': None, 'House 2': 1.0, 'House 3': 0.5} are correctly reported
 
+  """
+  Feedback required:
+  Earlied PAB was contiuously matching and announcing trade till it has been exhausted, but now it simply forwards the
+  recommended pair and rematch the derivative bids/offers in the next tick, due to which there are chances of unmatched
+  loads at the last tick leading to reduction in SS/SC
+  """
   Scenario: D3ASIM-2103, self sufficiency reports correct values on levels higher than houses with energy deficit for PAB
      Given we have a scenario named kpi.d3asim_2103_nopv1_2_sided_pab
      And d3a is installed
      When we run the d3a simulation on console with kpi.d3asim_2103_nopv1_2_sided_pab for 24 hrs (30, 30)
-     Then self_sufficiency of {'Grid': 1.0, 'Community': 0.833, 'House 1': 0.0, 'House 2': 0.5, 'House 3': 1.0} are correctly reported
-     And self_consumption of {'Grid': 1.0, 'Community': 1.0, 'House 1': None, 'House 2': 1.0, 'House 3': 0.5} are correctly reported
+     Then self_sufficiency of {'Grid': 1.0, 'Community': 0.833, 'House 1': 0.0, 'House 2': 0.25, 'House 3': 1.0} are correctly reported
+     And self_consumption of {'Grid': 1.0, 'Community': 1.0, 'House 1': None, 'House 2': 0.5, 'House 3': 0.5} are correctly reported
```

Comment:
So the self consumption increased in House 2, do you have an explanation for that?

Reply:
Same that now [seller_id](https://github.com/gridsingularity/d3a/commit/d7ddb9f38cab1706fd2e98791f112b42378d1291#diff-e504bd14031965b45bf3234fa47da69d52b61bc12220260f1a82a64aae8784b0R155) is properly tracked.

Reply:
Sorry, i miswrote. Self_sufficiency and self_consumption decrease for House 2 with your changes (reafctoring the matching algorithms). You blame the unmatched loads due to the lack of continues matching for this?

Reply:
> You blame the unmatched loads due to the lack of continues matching for this?

Yes, that's my observation/ understanding so far and dropped a doc-string comment for other to gain attention while reviewing

Reply:
I would suggest to give it another look to be sure that this is really the reason. 
Will review again, once other comment on this or just say: This is fine.

Reply:
Okay, after double checking House2's load isn't getting matched to its own PV for some market as House1's Load too bidding at the same rate, and then for the next tick House2's PV energy is already sold leading to reduce selfsufficiency/consumption at House2 level

Reply:
Thanks for explaining. If everybody is aware of this change and agrees, then all good.

Reply:
Agreed and I think this is an unavoidable aftermath of decoupling the matching from the clearing processes. For the time being though (until the real API is added) in order to satisfy the tests I would add a while loop that retries the matching and clearing until there are no further matches, similar to what the 2-sided PAB market did. I think it is worth to add it, not just for satisfying the tests, but also to satisfy the collaborations that we used in the Hackathon, which depended on this behavior.

Reply:
Adapted as per last comment https://github.com/gridsingularity/d3a/pull/1121/commits/a30de754640733711a85fe07861968887db0e4ac